samples/01_basic_ping_bot/Program.cs

```diff
@@ -14,23 +14,31 @@ namespace _01_basic_ping_bot
     // - Here, under the 02_commands_framework sample
     // - https://github.com/foxbot/DiscordBotBase - a bare-bones bot template
     // - https://github.com/foxbot/patek - a more feature-filled bot, utilizing more aspects of the library
-    class Program
+    class Program : IDisposable
     {
-        private DiscordSocketClient _client;
+        private readonly DiscordSocketClient _client;
 
         // Discord.Net heavily utilizes TAP for async, so we create
         // an asynchronous context from the beginning.
         static void Main(string[] args)
-            => new Program().MainAsync().GetAwaiter().GetResult();
+        {
+            using (var program = new Program())
+                program.MainAsync().GetAwaiter().GetResult();
+        }
 
-        public async Task MainAsync()
+        public Program()
```

Comment:
In addition I would advice against doing this kind of work in the constructor. Despite it causing confusion or rather being "untypically". Complex work in constructors should be avoided because of awkward behaviour when throwing exceptions. That is exactly why creational patterns exist. 
In this case there (probably) will never be an exception thrown. But for readability reasons I would just move that into MainAsync and be done with it.